bravado/http_future.py

```diff
@@ -165,10 +166,11 @@ def response(self, timeout=None, fallback_result=None, exceptions_to_catch=FALLB
             exc_info = list(sys.exc_info()[:2])
             exc_info.append(traceback.format_exc())
             if (
-                fallback_result and self.operation
+                fallback_result is not SENTINEL
+                and self.operation
                 and not self.operation.swagger_spec.config['bravado'].disable_fallback_results
             ):
-                swagger_result = fallback_result(e)
+                swagger_result = fallback_result(e) if callable(fallback_result) else fallback_result
```

Comment:
To me `callable` in general looks a good fit, as bravado-core models or primitives (ie.` int`, `bool`, etc) are not callable.

I do think that the approach could still lead to weird behaviours in case of `format` is used.
The scenario that could lead to issues could be if the bravado client has a custom format that returns callable objects
```python
def f_raise():
    raise ValueError()
def foo(*args, **kwargs): return 1
def bar(*args, **kwargs): return 2
_FUNCTIONS = {f.__name: f for f in [foo, bar]}

format = SwaggerFormat(
    format='function-format',
    to_wire=lambda func: func.__name__,
    to_python=lambda func_string: _FUNCTIONS.get(func_string, f_raise),
    validate=lambda *args, **kwargs: None
)
 ```

This is a case that _theoretically_ is supported even if I don't know if currently users are using it and/or why should someone do something like that.

So the question is: do we want to support such case?
If yes then probably `callable` is not a good fit and most probably we have no real way to determine if `fallback_result` is the function to call in case of exception or the result that should be returned
If no then `callable` is a good fit (eventually documentation should state that issue).

Reply:
Can you provide an actual example of what such a format would do? Otherwise I wouldn't worry about it.

Reply:
An example is available on https://gist.github.com/macisamuele/846c27c44569e1e06b513a82ee9f8070

We discussed, in person, about it and we agreed that such use case seems mostly a "misuse" of 
`SwaggerFormat` and so we should not be worried about it.